source/gfoidl.Base64/Internal/Base64Encoder/Base64Encoder.Decode.cs

```diff
@@ -377,8 +377,9 @@ private static bool Avx2Decode<T>(ref T src, ref byte destBytes, int sourceLengt
                 Vector256<sbyte> loNibbles = Avx2.And(str, mask2F);
                 Vector256<sbyte> hi        = Avx2.Shuffle(lutHi, hiNibbles);
                 Vector256<sbyte> lo        = Avx2.Shuffle(lutLo, loNibbles);
+                Vector256<sbyte> zero      = Avx.SetZeroVector256<sbyte>();
 
-                if (!Avx.TestZ(lo, hi))
+                if (Avx2.MoveMask(Avx2.CompareGreaterThan(Avx2.And(lo, hi), zero)) != 0)
```

Comment:
`MoveMask` is actually faster than `TestZ`.

```c#
using System;
using System.Runtime.Intrinsics;
using System.Runtime.Intrinsics.X86;
using BenchmarkDotNet.Attributes;

#if !DEBUG
using BenchmarkDotNet.Running;
#endif

namespace ConsoleApp3
{
    class Program
    {
        static void Main(string[] args)
        {
            var bench = new Bench();
            bool b1   = bench.TestZ();
            bool b2   = bench.MoveMask();

            Console.WriteLine(b1);
            Console.WriteLine(b2);

#if !DEBUG
            BenchmarkRunner.Run<Bench>();
#endif
        }
    }

    //[ShortRunJob]
    public class Bench
    {
        private static readonly Vector256<sbyte> _a;
        private static readonly Vector256<sbyte> _b;
        //---------------------------------------------------------------------
        static Bench()
        {
            _a = Avx.SetAllVector256<sbyte>(-42);
            _b = Avx.SetAllVector256<sbyte>(42);
        }
        //---------------------------------------------------------------------
        [Benchmark(Baseline = true)]
        public bool TestZ() => !Avx.TestZ(_a, _b);
        //---------------------------------------------------------------------
        [Benchmark]
        public bool MoveMask()
        {
            Vector256<sbyte> zero = Avx.SetZeroVector256<sbyte>();
            var and               = Avx2.And(_a, _b);
            var cmp               = Avx2.CompareGreaterThan(and, zero);
            int mask              = Avx2.MoveMask(cmp);

            return mask != 0;
        }
    }
}
```

`TestZ` (shortened, addresses are simplified):
```asm
       C4E17D104008         vmovupd  ymm0, ymmword ptr[rbx]
       C4E27D174008         vptest   ymm0, ymmword ptr[rcx]
       0F94C0               sete     al
       0FB6C0               movzx    rax, al
       85C0                 test     eax, eax
       0F94C0               sete     al
       0FB6C0               movzx    rax, al

G_M49958_IG03:
       C5F877               vzeroupper 
       C3                   ret      
```

`MoveMask` (shortened, addresses are simplified):
```asm
       C4E17DEFC0           vpxor     ymm0, ymm0, ymm0
       C4E17D104808         vmovupd   ymm1, ymmword ptr[rbx]
       C4E175DB4808         vpand     ymm1, ymm1, ymmword ptr[rcs]
       C4E17564C0           vpcmpgtb  ymm0, ymm1, ymm0
       C4E17DD7C0           vpmovmskb eax, ymm0
       85C0                 test      eax, eax
       0F95C0               setne     al
       0FB6C0               movzx     rax, al

G_M42351_IG03:
       C5F877               vzeroupper 
       C3                   ret      
```

``` ini

BenchmarkDotNet=v0.11.3, OS=Windows 10.0.17134.407 (1803/April2018Update/Redstone4)
Intel Core i7-7700HQ CPU 2.80GHz (Kaby Lake), 1 CPU, 8 logical and 4 physical cores
Frequency=2742187 Hz, Resolution=364.6724 ns, Timer=TSC
.NET Core SDK=3.0.100-preview-009765
  [Host]   : .NET Core 3.0.0-preview-27117-01 (CoreCLR 4.6.27116.05, CoreFX 4.7.18.56608), 64bit RyuJIT
  ShortRun : .NET Core 3.0.0-preview-27117-01 (CoreCLR 4.6.27116.05, CoreFX 4.7.18.56608), 64bit RyuJIT

Job=ShortRun  IterationCount=3  LaunchCount=1  
WarmupCount=3  

```
|   Method |      Mean |    Error |    StdDev | Ratio | RatioSD |
|--------- |----------:|---------:|----------:|------:|--------:|
|    TestZ | 0.7508 ns | 1.130 ns | 0.0619 ns |  1.00 |    0.00 |
| MoveMask | 0.5184 ns | 1.192 ns | 0.0653 ns |  0.69 |    0.07 |

Reply:
Raised https://github.com/dotnet/coreclr/issues/21247, so that codegen can be improved in the JIT.